pkgs/applications/editors/vim/plugins/neovim-require-check-hook.sh

```diff
@@ -1,24 +1,117 @@
 #shellcheck shell=bash
-# Setup hook for checking whether Python imports succeed
+# Setup hook for checking whether Lua imports succeed
 echo "Sourcing neovim-require-check-hook.sh"
 
-neovimRequireCheckHook () {
-    echo "Executing neovimRequireCheckHook"
+# Discover modules automatically if nvimRequireCheck is not set
+discover_modules() {
+    echo "Running module discovery in source directory..."
+
+    declare -A modules # Use associative array for unique module names
+    while IFS= read -r lua_file; do
+        if [[ "$lua_file" =~ lua/(.*)/init\.lua$ ]]; then
+            modules["${BASH_REMATCH[1]}"]=1 # Capture "name" from `lua/name/init.lua`
+        elif [[ "$lua_file" =~ /([^/]+)\.lua$ ]]; then
+            modules["${BASH_REMATCH[1]}"]=1 # Capture "name" from `name.lua`
```

Comment:
This captures `lua/myplugin/utils.lua` as `utils` instead of `myplugin.utils`. Could you add a check if the file is in `lua/`, then capture the full path as module name? Also, would be great to disable `stop_on_first_success` for such cases